src/github.com/getlantern/flashlight/geolookup/geolookup.go

```diff
@@ -0,0 +1,94 @@
+package geolookup
+
+import (
+	"encoding/json"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/getlantern/flashlight/ui"
+	"github.com/getlantern/geolookup"
+	"github.com/getlantern/golog"
+)
+
+const (
+	messageType = `GeoLookup`
+
+	sleepTime = time.Second * 10
+)
+
+var (
+	log = golog.LoggerFor("geolookup-flashlight")
+
+	service     *ui.Service
+	lookupMutex sync.Mutex
+	lookupData  *geolookup.City
+)
+
+func getUserGeolocationData() *geolookup.City {
+	lookupMutex.Lock()
+	defer lookupMutex.Unlock()
+
+	var err error
+
+	if lookupData != nil {
+		// We already looked up IP's information.
+		return lookupData
+	}
+
+	for {
+		if !geolookup.UsesDefaultHTTPClient() {
```

Comment:
So if this is false, we'll never do a geo lookup?  Why even bother starting the service?